common/workunit/workunit.cpp

```diff
@@ -13636,12 +13642,17 @@ void executeThorGraph(const char * graphName, IConstWorkUnit &workunit, const IP
         runningTimeLimit = runningTimeLimit ? runningTimeLimit : INFINITE;
 
         std::list<WUState> expectedStates = { WUStateRunning, WUStateWait };
+        unsigned __int64 blockedTime = 0;
         for (unsigned i=0; i<2; i++)
         {
             WUState state = waitForWorkUnitToComplete(wuid, timelimit*1000, expectedStates);
             DBGLOG("Got state: %s", getWorkunitStateStr(state));
             if (WUStateWait == state) // already finished
+            {
+                if (!blockedTime)
```

Comment:
worth a comment here I think, along the lines of:
// wuid may have spent time in blocked state, but then transitioned to wait state quickly such that this code did not see it's running state.